Improve log output of the `testAll` task

szeiger opened the pull request:
It’s a lot of code for little benefit but makes the output more useful
when test tasks fail. Unfortunately there doesn’t seem to be any way to
get the `summary` reported by a test framework at this point. The
arguments of `toTask` for InputTasks with applied arguments have also
been lost, so we keep track of the commands separately.

szeiger committed:
Improve log output of the `testAll` task

It’s a lot of code for little benefit but makes the output more useful
when test tasks fail. Unfortunately there doesn’t seem to be any way to
get the `summary` reported by a test framework at this point. The
arguments of `toTask` for InputTasks with applied arguments have also
been lost, so we keep track of the commands separately.

adriaanm commented:
Every improvement to test output is worth it to me! User-friendliness to new contributors as well as keeping our own eyes from bleeding scanning through unnecessary punctuation!

szeiger committed:
Split “partest run” off from “partest pos neg jvm”

We’ve seen several OOM failures in “run” tests lately. Maybe going back
to more separate partest calls will help. Now that everything is
launched from the same sbt instance and test results are always
accumulated, this should not have any negative impact on build
performance or usability.

adriaanm commented:
LGTM!
